File: samples/java/src/main/java/co/windly/limbosample/presentation/main/settings/SettingsFragment.java
package co.windly.limbosample.presentation.main.settings;

import co.windly.limbosample.R;
import co.windly.limbosample.presentation.base.fragment.base.BaseFragment;

public class SettingsFragment extends BaseFragment<SettingsView, SettingsPresenter> implements SettingsView {

  //region Fragment

  public static SettingsFragment createInstance() {
    return new SettingsFragment();
  }

  //endregion

  //region Ui

  @Override
  protected int getLayout() {
    return R.layout.fragment_settings;
  }

  //endregion
}

File: samples/kotlin/src/main/java/co/windly/limbosample/presentation/main/settings/SettingsFragment.kt
package co.windly.limbosample.presentation.main.settings

import co.windly.limbosample.R
import co.windly.limbosample.presentation.base.fragment.base.BaseFragment

class SettingsFragment : BaseFragment<SettingsView, SettingsPresenter>(), SettingsView {

  //region Fragment

  companion object {

    fun createInstance(): SettingsFragment =
      SettingsFragment()
  }

  //endregion

  //region Ui

  override val layout: Int
    get() = R.layout.fragment_settings

  //endregion
}
